odoo/fields.py

```diff
@@ -2392,100 +2392,71 @@ def read(self, records):
             cache.set(record, self, tuple(group[record.id]))
 
     def create(self, record_values):
-        if not record_values:
+        self._write(record_values)
+
+    def write(self, records, value):
+        self._write([(records, value)])
+
+    def _write(self, records_commands_list):
+        # records_commands_list = [(records, commands), ...]
+        if not records_commands_list:
             return
 
-        model = record_values[0][0]
+        model = records_commands_list[0][0].browse()
         comodel = model.env[self.comodel_name].with_context(**self.context)
         inverse = self.inverse_name
-        vals_list = []                  # vals for lines to create in batch
 
-        def flush():
-            if vals_list:
-                comodel.create(vals_list)
-                vals_list.clear()
+        to_create = []                  # line vals to create
+        to_delete = []                  # line ids to delete
+        to_relink = {}                  # lines to relink {line_id: record_id}
 
-        def drop(lines):
+        def unlink(line_ids):
             if comodel._fields[inverse].ondelete == 'cascade':
-                lines.unlink()
+                to_delete.extend(line_ids)
             else:
-                lines.write({inverse: False})
+                to_relink.update(dict.fromkeys(line_ids, False))
```

Comment:
Doesn't this mean that trying to unlink records of a comodel whose ondelete == 'null' will not be done in batch? i.e. it will still be slower than performing `.unlink()` on the recordset?

Reply:
It will be done in batch. With `ondelete='set null'`, the unlink command will assign the inverse many2one field to `False`, thanks to the `to_relink` dict. And `to_relink` processes stuff in batch.